File: seeker/snippet/Message.java
//date: 2022-12-08T16:59:51Z
//url: https://api.github.com/gists/59060d8a567994f4f60bc34e43b0ddd4
//owner: https://api.github.com/users/triangle1984

package com.example.websocketclient;

import lombok.Data;


/**
 * @author Andrey Markov
 * @version 1.0 27.06.2021
 */
@Data
public class Message {
    String from;
    String to;
    String text;

    public Message() {

    }

    public Message(String from) {
        this.from = from;
    }

    public Message(String from, String text, String to) {
        this.from = from;
        this.to = to;
        this.text = text;
    }
}

File: seeker/snippet/message.go
//date: 2023-07-20T17:02:14Z
//url: https://api.github.com/gists/a920dcfea6cca057e517443c8a9423c0
//owner: https://api.github.com/users/delveper

package notif

import (
	"bytes"
	"fmt"
	"text/template"
)

// ExchangeRateContent responsible for creating exchange rate content for sending messages.
type ExchangeRateContent struct{ *template.Template }

func NewExchangeRateContent(tmpl *template.Template) *ExchangeRateContent {
	return &ExchangeRateContent{tmpl}
}

func (c *ExchangeRateContent) CreateMessage(md *MetaData) (*Message, error) {
	tos := make([]string, len(md.subss))

	for i := range md.subss {
		tos[i] = md.subss[i].Subscriber.Address.String()
	}

	var buf bytes.Buffer
	if err := c.ExecuteTemplate(&buf, "subject", md); err != nil {
		return nil, fmt.Errorf("executing body template: %w", err)
	}

	subj := buf.String()

	buf.Reset()
	if err := c.ExecuteTemplate(&buf, "body", md); err != nil {
		return nil, fmt.Errorf("executing subject template: %w", err)
	}

	body := buf.String()

	msg := Message{
		To:      tos,
		Subject: subj,
		Body:    body,
	}

	return &msg, nil
}